offer-manager-domain/src/main/groovy/life/qbic/business/offers/QuantityDiscount.groovy

```diff
@@ -0,0 +1,198 @@
+package life.qbic.business.offers
+
+import java.util.function.BiFunction
+
+/**
+ * <b>Function for quantity discount</b>
+ *
+ * <p>Applies a quantity discount to the discountable price depending on the number of samples.</p>
+ *
+ * @since 1.1.0
+ */
+class QuantityDiscount implements BiFunction<Integer, Double, Double> {
+
+    /**
+     * Applies a quantity discount to the discountable price depending on the number of samples.
+     *
+     * @param sampleCount the number of samples
+     * @param discountablePrice the price that should be discounted
+     * @return the function the discountablePrice with discounts applied to it
+     * @since 1.1.0
+     */
+    @Override
+    Double apply(Integer sampleCount, Double discountablePrice) {
+        return getDiscountFactor(sampleCount) * discountablePrice
+    }
+
+    private static double getDiscountFactor(int sampleCount) {
+        double discountFactor
```

Comment:
you can let the method fail here, if the sampleCount is <= 0. If its >= 1000 return the minimum discount factor.

Then you can evaluate the other options